etna/core/mixins.py

```diff
@@ -26,6 +28,26 @@ def __repr__(self):
                 args_str_representation += f"{arg} = {repr(value)}, "
         return f"{self.__class__.__name__}({args_str_representation})"
 
+    def to_dict(self):
+        """Collect all information about etna object in dict."""
+        init_args = inspect.signature(self.__init__).parameters
+        params = {}
+        for arg in init_args.keys():
+            value = self.__dict__[arg]
+            if isinstance(value, BaseMixin):
+                params[arg] = value.to_dict()
+            elif isinstance(value, BaseEstimator):
+                params[arg] = {}
+                params[arg]["_target_"] = value.__class__
+                model_parameters = value.get_params()
+                params[arg].update(model_parameters)
+            else:
+                params[arg] = value
```

Comment:
If class is not from our known family. We should pass only target field for them.

Because `self.__dict__[arg]` will throw errors for unknown objects in case all arguments are passing to private attributes.